src/query-builder/UpdateQueryBuilder.ts

```diff
@@ -370,7 +370,7 @@ export class UpdateQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                     //
                     let value = column.getEntityValue(valuesSet);
                     if (column.referencedColumn && value instanceof Object) {
-                        value = column.referencedColumn.getEntityValue(value);
+                        value = column.referencedColumn.getEntityValue(valuesSet);
```

Comment:
I don't know this is destructive change or not.
But I passed unit tests (on my local env).
Does these unit tests cover all cases?

Reply:
believe me or not but there are at least million of use cases in orm. So no, it does not cover all million cases :) but change is most probably destructive.